dotCMS/src/main/java/com/dotmarketing/fixtask/tasks/FixTask00095DeleteOrphanRelationships.java

```diff
@@ -0,0 +1,199 @@
+                        FixTask00095DeleteOrphanRelationships.class,
+                        "There was a problem during DeleteOrphanRelationships",
+                        e);
+                HibernateUtil.rollbackTransaction();
+                FixAssetsProcessStatus.setActual(-1);
+            } catch (Exception e) {
+                Logger.debug(
+                        FixTask00095DeleteOrphanRelationships.class,
+                        "There was a problem during DeleteOrphanRelationships",
+                        e);
+                HibernateUtil.rollbackTransaction();
+                FixAssetsProcessStatus.setActual(-1);
+            } finally {
+                FixAssetsProcessStatus.stopProgress();
+            }
+        }
+        return returnValue;
+    }
+
+    @Override
+    public List<Map<String, String>> getModifiedData() {
+        if (modifiedData.isEmpty()) {
+            XStream xstreamObj = new XStream(new DomDriver());
+            LocalDate date = LocalDate.now();
+            SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy_HH-mm-ss");
+            String lastmoddate = sdf.format(date);
+            File writingObj = null;
+            if (!new File(ConfigUtils.getBackupPath() + File.separator
+                    + "fixes").exists()) {
+                new File(ConfigUtils.getBackupPath() + File.separator + "fixes")
+                        .mkdirs();
+            }
+            writingObj = new File(ConfigUtils.getBackupPath() + File.separator
+                    + "fixes" + java.io.File.separator + lastmoddate + "_"
+                    + "FixTask00095DeleteOrphanRelationships" + ".xml");
+
+            BufferedOutputStream bufferedOutObj = null;
+            try {
+                bufferedOutObj = new BufferedOutputStream(new FileOutputStream(writingObj));
+            } catch (FileNotFoundException e) {
+                Logger.error(this, "Could not write to Fix Task status file.");
+            }
+            xstreamObj.toXML(modifiedData, bufferedOutObj);
+        }
+        return modifiedData;
+    }
+
+    @Override
+    public boolean shouldRun() {
+        Logger.debug(CMSMaintenanceFactory.class, "Running query for relationships: "
+                + VERIFICATION_QUERY);
+        DotConnect dc = new DotConnect();
+        dc.setSQL(VERIFICATION_QUERY);
+        List<Map<String, Object>> inodesInStructure = new ArrayList<>();
+        
+        try {
+            inodesInStructure = dc.loadObjectResults();
+        } catch (DotDataException e) {
+            Logger.error(this, e.getMessage(), e);
+        }
+        
+        int total = (inodesInStructure != null && !inodesInStructure.isEmpty()) ? 
+                inodesInStructure.size() : 0;
+                
+        Logger.debug(CMSMaintenanceFactory.class,
+                "Found " + total + " invalid inodes.");
+        
+        FixAssetsProcessStatus.setTotal(total);
+        return total > 0 ? true : false;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)